### DISCUSSION TITLE
Using multiple Field in the type annotations for the same field

### DISCUSSION THREAD

**Discussion by imneedle** [fastapi/sqlmodel#947] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [x] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from sqlalchemy import inspect
from sqlmodel import SQLModel, Field


def IndexField() -> Field:
    return Field(index=True)


def UniqueField() -> Field:
    return Field(unique=True)


def ForeignKeyField(foreign_key: str) -> Field:
    return Field(foreign_key=foreign_key)


def PrimaryKeyField() -> Field:
    return Field(primary_key=True)


class Test(SQLModel, table=True):
    id: Annotated[int | None, PrimaryKeyField()]
    x: Annotated[int, UniqueField(), IndexField()]
    y: Annotated[int | None, ForeignKeyField("subtest.id"), IndexField()]


class SubTest(SQLModel, table=True):
    id: Annotated[int | None, PrimaryKeyField()]


mapper = inspect(Test)
for column in mapper.tables[0].columns:
    print(f"{column=!r}")
    print(f"{column.primary_key=}")
    print(f"{column.unique=}")
    print(f"{column.index=}")
    print(f"{column.foreign_keys=}")
```

### Description

Hello. I am trying to set up my database using `SQLModel`.

Doing so, I am getting tired of some of the repetition when setting `foreign_key`, `index` or `unique` on the `Field` annotations.

I would like to be able to reach something like this:

```py
from typing import Annotated

from sqlmodel import SQLModel

from .fields import PrimaryKeyField, IndexField, UniqueField, ForeignKeyField

class Test(SQLModel, table=True):
    id: Annotated[int | None, PrimaryKeyField()]
    x: Annotated[int, UniqueField(), IndexField()]
    y: Annotated[int | None, ForeignKeyField("subtest.id")]
```

And the `fields.py` field would look like:

```py
from sqlmodel import Field


def IndexField() -> Field:
    return Field(index=True)


def UniqueField() -> Field:
    return Field(unique=True)


def ForeignKeyField(foreign_key: str) -> Field:
    return Field(foreign_key=foreign_key)


def PrimaryKeyField() -> Field:
    return Field(primary_key=True)
```

When testing out, I used `sqlalchemy.inspect` on the model, to see if the constraints were applied. The output shows that it didn't work:

```py
from sqlalchemy import inspect

from .test import Test

mapper = inspect(Test)
for column in mapper.tables[0].columns:
    print(f"{column=!r}")
    print(f"{column.primary_key=}")
    print(f"{column.unique=}")
    print(f"{column.index=}")
    print(f"{column.foreign_keys=}")
```

### Operating System

Linux

### SQLModel Version

0.0.18

### Python Version

Python 3.12.3

### Additional Context

I would like to add that this composition of `Field` seems to work in Pydantic...

```py
from typing import Annotated

from pydantic import Field, BaseModel


class Test(BaseModel):
    x: Annotated[int, Field(ge=0), Field(lt=100)]


print(Test.model_fields)
```

Which would output the following:

```
{'x': FieldInfo(annotation=int, required=True, metadata=[Ge(ge=0), Lt(lt=100)])}
```

Additionally, I tried using the same approach in SQLModel, and still no luck:

```py
y: Annotated[int | None, Field(unique=True), Field(index=True), Field(foreign_key="subtest.id")]
```

**Comment by joachimhuet** [comment]
You should look into that thread: #52 
When `table=True` annotations and therefore validation is deactivated.

**Comment by alexandre-piccin** [comment]
Any updates on this?
